sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -484,6 +493,10 @@ def fit_binary(
     return coef, intercept, n_iter_
 
 
+def _get_plain_sgd_function(x_dtype):
+    return _plain_sgd64 if x_dtype == np.float64 else _plain_sgd32
```

Comment:
Python implementations must have as little adherence to the Cython routines as possible.

Can you instead implement `_plain_sgd` in `sklearn/linear_model/_sgd_fast.pyx.tp` as a front-end dispatching on either `_plain_sgd64` or `_plain_sgd32`?

This can be done by implementing a Python function before the Tempita for-loop. As an example see this file's: https://github.com/scikit-learn/scikit-learn/blob/6f6b4bdd9c94ec2068e5dfcca05be259b888347a/sklearn/metrics/_pairwise_distances_reduction/_base.pyx.tp

Reply:
Do you mean I should have a function above the tempita code which checks the dtype and calls the respective function?

Reply:
Yes, in particular, the import line must not change.

Another I think good example are adaptors to BLAS: https://github.com/scikit-learn/scikit-learn/blob/fb9c5496894513ceb56486a38cb71909687c0888/sklearn/utils/_cython_blas.pyx